src/collapse/dtrace.rs

```diff
@@ -196,7 +196,7 @@ impl Folder {
         crossbeam::thread::scope(|scope| {
             // Spin up the threadpool / worker threads.
             let (tx_input, rx_input) = channel::bounded::<Option<Vec<u8>>>(2 * self.opt.nthreads);
-            let (tx_output, rx_output) = channel::bounded(self.opt.nthreads);
+            let (tx_output, rx_output) = channel::bounded::<io::Result<()>>(1);
```

Comment:
I think leaving this bound as `1` has the potential to deadlock. In particular, imagine that each thread encounters one broken stack. All but one block on their send to `tx_output`. Now imagine that the one that _didn't_ block encounters another broken stack. It will also block. At this point, no threads are reading from `rx_input`, so the main thread blocks forever, and there's a deadlock. The error channel should either be unbounded, or have a capacity of `self.opt.nthreads` with every thread using `break` instead of `continue` if they encounter an error.

Reply:
Since we only handle the _first_ error anyway, these threads may as well all just `break` if they ever encounter an error, in which case the bound should be `self.opt.nthreads`.

Reply:
That's why, in the children, I changed `tx_output.send` to `tx_output.try_send` - To avoid the deadlock you're talking about. We can't break on error because we need the children to continue to try to receive data on the input channel.

Reply:
Oh, I see what you mean -- I missed that. Yeah, `try_send` should be fine. Maybe leave a comment about why that's a `try_send`, and why we `continue` nonetheless?